packages/tools/src/tools/annotation/planarFreehandROITool/interpolatePoints.ts

```diff
@@ -0,0 +1,176 @@
+import { Types } from '@cornerstonejs/core';
+import { point } from '../../../utilities/math';
+import interpolateSegmentPoints from './interpolation/interpolateSegmentPoints';
+
+export function assertInterpolation(configuration: Record<any, any>): boolean {
+  return configuration?.interpolation?.enabled === true;
+}
+
+/**
+ * Tells whether two points are equal by proximity or not as far as interpolation goes.
+ */
+function isEqualByProximity(pointA, pointB) {
+  return point.distanceToPoint(pointA, pointB) < 0.001;
+}
+
+/**
+ * Tells whether two points are strictly equal or not as far as interpolation goes.
+ */
+function isEqual(pointA, pointB) {
+  return point.distanceToPoint(pointA, pointB) === 0;
+}
+
+/**
+ * Finds the indexes of points list and otherPoints list that points are identical.
+ */
+function findMatchIndexes(
+  points: Types.Point2[],
+  otherPoints: Types.Point2[]
+): [number, number] | undefined {
+  for (let i = 0; i < points.length; i++) {
+    for (let j = 0; j < otherPoints.length; j++) {
+      if (isEqual(points[i], otherPoints[j])) {
+        return [i, j];
+      }
+    }
+  }
+}
+/**
+ * Returns the following index value (on circular basis) of index param on the given direction.
+ */
+function followingIndex(
+  index: number,
+  size: number,
+  direction: number
+): number {
+  return (index + size + direction) % size;
+}
+
+/**
+ * Circular finding that returns the next index for two list where the criteria is met.
+ */
+function circularFindNextIndexBy(
+  pointContent: [number, number, Types.Point2[]],
+  otherPointContent: [number, number, Types.Point2[]],
+  criteria: (pointA: Types.Point2, pointB: Types.Point2) => boolean,
+  direction: number
+): [number, number] {
+  const [, indexDelimiter, points] = pointContent;
+  const [, otherIndexDelimiter, otherPoints] = otherPointContent;
+
+  const pointsLength = points.length;
+  const otherPointsLength = otherPoints.length;
+
+  let startIndex = pointContent[0];
+  let otherStartIndex = otherPointContent[0];
+
+  function validIndex(
+    _index: number,
+    _indexDelimiter: number,
+    _points: Types.Point2[]
+  ): boolean {
+    return _points[_index] && _index !== _indexDelimiter;
+  }
+
+  while (
```

Comment:
What are the return conditions when:
1. Neither initial point is valid initially
2. One of the indices becomes invalid during the iteration
3. The criteria is met?
Are these all the same expected values, or are there edge cases to be considered here?

Reply:
I have updated the function and its returning type to consider:
If return type is valid it should be [number,number] otherwise should be [undefined, undefined]